mypy/treetransform.py

```diff
@@ -486,25 +492,25 @@ def visit_type_alias_expr(self, node: TypeAliasExpr) -> TypeAliasExpr:
     def visit_newtype_expr(self, node: NewTypeExpr) -> NewTypeExpr:
         return NewTypeExpr(node.info)
 
-    def visit_namedtuple_expr(self, node: NamedTupleExpr) -> Node:
+    def visit_namedtuple_expr(self, node: NamedTupleExpr) -> NamedTupleExpr:
         return NamedTupleExpr(node.info)
 
-    def visit__promote_expr(self, node: PromoteExpr) -> Node:
+    def visit__promote_expr(self, node: PromoteExpr) -> PromoteExpr:
         return PromoteExpr(node.type)
 
-    def visit_temp_node(self, node: TempNode) -> Node:
+    def visit_temp_node(self, node: TempNode) -> TempNode:
         return TempNode(self.type(node.type))
 
-    def node(self, node: Node) -> Node:
-        new = node.accept(self)
+    def node(self, node: T) -> T:
+        new = cast(T, node.accept(self))
```

Comment:
Ow, it was going so well, but I really don't like the idea of needing this cast(). Is that because there's no selftype? Then you should at least add a big TODO -- or preferably first fix seltype (there's an issue for that!).